crates/testing/tests/combined_network.rs

```diff
@@ -39,28 +82,213 @@ async fn test_combined_network() {
         ..TestMetadata::default_multiple_rounds()
     };
 
+    let mut all_nodes = vec![];
+    for node in 0..metadata.total_nodes {
+        all_nodes.push(ChangeNode {
+            idx: node,
+            updown: UpDown::NetworkDown,
+        });
+    }
+
+    metadata.spinning_properties = SpinningTaskDescription {
+        node_changes: vec![(Duration::new(4, 0), all_nodes)],
+    };
+
     metadata
         .gen_launcher::<TestTypes, CombinedImpl>()
         .launch()
         .run_test()
-        .await
+        .await;
+
+    // async_std needs time to spin down the handler
+    #[cfg(async_executor_impl = "async-std")]
+    async_std::task::sleep(Duration::from_secs(4)).await;
 }
 
-// stress test for web server with libp2p
+// A run where the webserver crashes partway through
+// and then comes back up
+#[cfg_attr(
+    async_executor_impl = "tokio",
+    tokio::test(flavor = "multi_thread", worker_threads = 2)
+)]
+#[cfg_attr(async_executor_impl = "async-std", async_std::test)]
+#[instrument]
+async fn test_combined_network_reup() {
+    async_compatibility_layer::logging::setup_logging();
+    async_compatibility_layer::logging::setup_backtrace();
+    let mut metadata: TestMetadata = TestMetadata {
+        timing_data: TimingData {
+            round_start_delay: 25,
+            next_view_timeout: 10000,
+            start_delay: 120000,
+
+            ..Default::default()
+        },
+        overall_safety_properties: OverallSafetyPropertiesDescription {
+            num_successful_views: 35,
+            ..Default::default()
+        },
+        // allow more time to pass in CI
+        completion_task_description: CompletionTaskDescription::TimeBasedCompletionTaskBuilder(
+            TimeBasedCompletionTaskDescription {
+                duration: Duration::from_millis(1_200_000),
```

Comment:
nit can we use `from_seconds` for large numbers like this

Reply:
Ah, good catch. I forgot to change this back to 60 seconds after testing